src/coreclr/vm/interpexec.cpp

```diff
@@ -6,6 +6,57 @@
 #include "threads.h"
 #include "gcenv.h"
 #include "interpexec.h"
+#include "callstubgenerator.h"
+
+void InvokeCompiledMethod(MethodDesc *pMD, int8_t *pArgs, int8_t *pRet)
+{
+    CONTRACTL
+    {
+        THROWS;
+        MODE_ANY;
+        PRECONDITION(CheckPointer(pMD));
+        PRECONDITION(CheckPointer(pArgs));
+        PRECONDITION(CheckPointer(pRet));
+    }
+    CONTRACTL_END
+
+    CallStubGenerator callStubGenerator;
+    CallStubHeader *pHeader = pMD->GetCallStub();
+    if (pHeader == NULL)
+    {
+        AllocMemTracker amTracker;
+        {
+            GCX_PREEMP();
+            pHeader = callStubGenerator.GenerateCallStub(pMD, &amTracker);
+        }
+
+        if (pHeader == NULL)
+        {
+            // allocating the header has failed due to OOM
+            COMPlusThrowOM();
+        }
+
+        HRESULT hr = pMD->SetCallStub(pHeader);
```

Comment:
Can this be changed to just return `bool`?

Reply:
I've done it the same way like the other methods that modifies the `MethodDescData` - the `SetMethodDescVersionState`, do we want to diverge this new one and let it throw instead?

Reply:
Looking at the code, `SetMethodDescVersionState` is special and tries to go to extra length to avoid throwing exceptions (no sure why). `SetCallStub` does not need to be special like that.

Other similar method that operates on `MethodDescData` - `EnsureTemporaryEntryPointCore` - is on a regular error handling plan.

Reply:
Ok